google/cloud/error.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/error.h"
+#include <iostream>
+#include <mutex>
+
+namespace google {
+namespace cloud {
+inline namespace GOOGLE_CLOUD_CPP_NS {
+namespace {
+
+class TerminateFunction {
+ public:
+  TerminateFunction(TerminateHandler&& f) {
+    std::lock_guard<std::mutex> l(m_);
```

Comment:
nit: you do not need to grab the mutex in the constructor, but it works... meh?  No action needed.

Reply:
True, looks like I didn't code for too long...